fs-index/src/index.rs

```diff
@@ -1,1120 +1,491 @@
-            .root
-            .to_owned()
-            .join(ARK_FOLDER)
-            .join(INDEX_PATH);
-
-        let ark_dir = index_path.parent().unwrap();
-        fs::create_dir_all(ark_dir)?;
+        Ok(())
+    }
 
-        let mut file = File::create(index_path)?;
+    /// Get resources by their ID
+    ///
+    /// Returns None if there is no resource with the given ID
+    ///
+    /// **Note**: This can return multiple resources with the same ID in case of
+    /// hash collisions or files with the same content
+    pub fn get_resources_by_id(
+        &self,
+        id: &Id,
+    ) -> Option<Vec<IndexedResource<Id>>> {
+        let mut resources = vec![];
+
+        let paths = self.id_to_paths.get(id)?;
+        for path in paths {
+            let resource = self.path_to_resource.get(path)?;
+            let resource = IndexedResource::new(
+                resource.id.clone(),
+                path.clone(),
+                resource.last_modified,
+            );
+            resources.push(resource);
+        }
 
-        let mut path2id: Vec<(&CanonicalPathBuf, &IndexEntry<Id>)> =
-            self.path2id.iter().collect();
-        path2id.sort_by_key(|(_, entry)| *entry);
+        Some(resources)
+    }
 
-        for (path, entry) in path2id.iter() {
-            log::trace!("[store] {} by path {}", entry.id, path.display());
+    /// Get a resource by its path
+    ///
+    /// Returns None if the resource does not exist
+    ///
+    /// **Note**: The path should be relative to the root path
+    pub fn get_resource_by_path<P: AsRef<Path>>(
+        &self,
+        path: P,
+    ) -> Option<IndexedResource<Id>> {
+        let resource = self.path_to_resource.get(path.as_ref())?;
+        let resource = IndexedResource::new(
+            resource.id.clone(),
+            path.as_ref().to_path_buf(),
+            resource.last_modified,
+        );
+        Some(resource)
+    }
 
-            let timestamp = entry
-                .modified
-                .duration_since(UNIX_EPOCH)
-                .map_err(|_| {
-                    ArklibError::Other(anyhow!("Error using duration since"))
-                })?
-                .as_millis();
+    /// Build a new index from the given root path
+    pub fn build<P: AsRef<Path>>(root_path: P) -> Result<Self> {
+        log::debug!("Building index at root path: {:?}", root_path.as_ref());
 
-            let path =
-                pathdiff::diff_paths(path.to_str().unwrap(), self.root.clone())
-                    .ok_or(ArklibError::Path(
-                        "Couldn't calculate path diff".into(),
-                    ))?;
+        let root_path = root_path.as_ref();
+        // Canonicalize the root path (only for unix-like systems)
+        // Note: On windows, canonicalization adds a prefix to the path
+        // which breaks the path comparison in the tests
+        #[cfg(target_family = "unix")]
+        let root_path = root_path.canonicalize()?;
```

Comment:
Do we simply skip canonicalization on Windows now?

Reply:
Please ignore this commit—it was just for troubleshooting. I’ll revert it in the next one. 
I didn’t want to make your life harder with force-pushing 😃